mypy/messages.py

```diff
@@ -992,6 +997,25 @@ def pretty_or(args: List[str]) -> str:
     return ", ".join(quoted[:-1]) + ", or " + quoted[-1]
 
 
+def invariance_notes(notes: List[str], arg_type: Type, expected_type: Type) -> List[str]:
+    """Explain that the type is invariant and give notes for how to solve the issue."""
+    if (arg_type.type.fullname() == 'builtins.list' and
```

Comment:
This will probably fail self-check, since ``Type`` (annotation for ``arg_type``) has no attribute ``type``. Use ``Instance`` annotation and ``if isinstance(..., Instance )`` at call site.